### PR TITLE
Schedule workflows

### PR DESCRIPTION
**Pull request by pacospace**
Signed-off-by: Francesco Murdaca <fmurdaca@redhat.com>

## Related Issues and Dependencies

Fixes: https://github.com/thoth-station/investigator/issues/101

Fixes: https://github.com/thoth-station/solver/issues/584

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

Add logic to schedule solvers, revsolver, security indicators when UnresolvedPackageMessage is received

### CODE REVIEW

File: thoth/investigator/investigate_unresolved_package.py [hunk 1]
```diff
@@ -120,67 +122,226 @@ def parse_unresolved_package_message(unresolved_package: MessageBase) -> None:
...
+
+def learn_about_security(is_present: bool, package_name: str, indexes: List[str], package_version: str):
+    """Learn about security for Package Version Index."""
+    if is_present:
+        is_si_analyzed = si_aggregated_python_package_version_exists(
+            package_name=package_name, package_version=package_version, index_url=index_url
+        )
+
+        if is_si_analyzed:
+            return is_si_analyzer_scheduled
```
[the remark this answers is not in the document]
Comment (pacospace) [reply thoth/investigator/investigate_unresolved_package.py:280]: I might have mixed variables name :D Thanks for spotting this @KPostOffice , I will correct it :)

File: thoth/investigator/investigate_unresolved_package.py [hunk 2]
```diff
@@ -120,67 +122,226 @@ def parse_unresolved_package_message(unresolved_package: MessageBase) -> None:
...
+        _LOGGER.debug("consider index %r", index_url)
+        source = Source(index_url)
+
+        versions = []
+
+        try:
+            versions = source.get_package_versions(package_name)
+
+        except Exception as exc:
+            _LOGGER.exception(str(exc))
```
[the remark this answers is not in the document]
Comment (pacospace) [reply thoth/investigator/investigate_unresolved_package.py]: I just changed it to warning indeed, thanks!

File: thoth/investigator/investigate_unresolved_package.py [hunk 3]
```diff
@@ -120,67 +122,226 @@ def parse_unresolved_package_message(unresolved_package: MessageBase) -> None:
     requested_indexes: Optional[List[str]] = unresolved_package.index_url
     solver = unresolved_package.solver
 
+    openshift = OpenShift()
+
     graph = GraphDatabase()
     graph.connect()
 
+    if not package_version:
+        _LOGGER.debug("consider index %r", index_url)
```
[the remark this answers is not in the document]
Comment (pacospace) [reply thoth/investigator/investigate_unresolved_package.py]: I moved this logic below :) Thanks!

File: thoth/investigator/investigate_unresolved_package.py [hunk 4]
```diff
@@ -120,67 +112,269 @@ def parse_unresolved_package_message(unresolved_package: MessageBase) -> None:
...
+    if not solver:
+        solvers: List[str] = openshift.get_solver_names()
+    else:
+        solvers = [solver]
+
+    # Check for which solver has not been solved and schedule solver workflow
+    are_solvers_scheduled = 0
+
+    for solver_name in solvers:
+        is_solved = graph.python_package_version_exists(
```
Comment (KPostOffice) [inline thoth/investigator/investigate_unresolved_package.py:229]: Don't we already have this from `is_present`?

**Pull request merged**